src/marqo/tensor_search/models/score_modifiers_object.py

```diff
@@ -1,65 +1,95 @@
-score_modifiers_object_schema = {
-    "$schema": "http://json-schema.org/draft-07/schema#",
-    "type": "object",
-    "properties": {
-        "multiply_score_by": {
-            "type": "array",
-            "minItems": 1,
-            "items": {
-                "type": "object",
-                "properties": {
-                    "field_name": {
-                        "type": "string",
-                        "not": {
-                            "const": "_id"
-                        }
-                    },
-                    "weight": {
-                        "type": "number",
-                        "default": 1
-                    }
-                },
-                "required": [
-                    "field_name"
-                ],
-                "additionalProperties": False
-            }
-        },
-        "add_to_score": {
-            "type": "array",
-            "minItems": 1,
-            "items": {
-                "type": "object",
-                "properties": {
-                    "field_name": {
-                        "type": "string",
-                        "not": {
-                            "const": "_id"
-                        }
-                    },
-                    "weight": {
-                        "type": "number",
-                        "default": 1
-                    }
-                },
-                "required": [
-                    "field_name"
-                ],
-                "additionalProperties": False
-            }
-        }
-    },
-    "anyOf": [
-        {
-            "required": [
-                "multiply_score_by"
-            ]
-        },
-        {
-            "required": [
-                "add_to_score"
-            ]
-        }
-    ],
-    "additionalProperties": False
-}
+from typing import List, Dict, Any, Optional, Literal
+import json 
+
+from pydantic import BaseModel, validator, ValidationError
+
+from marqo.errors import InvalidArgError
+
+class ScoreModifierValidationError(InvalidArgError):
+    def __init__(self, modifier: Dict[str, Any], message: str, link: str = None):
+        super().__init__(
+            link=link,
+            message=f"Error validating score_modifiers = `{modifier}`. Reason: \n{message} "
+            f"Please revise your score_modifiers based on the provided error."
+            f"\n Check `https://docs.marqo.ai/0.0.17/API-Reference/search/#score-modifiers` for more info."
+        )
+
+
+class ScoreModifierOperator(BaseModel):
+    field_name: str
+    weight: float = 1
+
+    class Config:
+        extra: str = "forbid"
+
+    @validator('field_name')
```

Comment:
This is true, although I think is additional validation (because ScoreModifierOperator.fields are validated at field creation time and when ScoreModifierOperator looks for the field name.